### PR TITLE
KYLIN-4205 configuration support set hive.intermediate-table-prefix

### PR DESCRIPTION
**Pull request by bigxiaochu**

### CODE REVIEW

File: core-cube/src/main/java/org/apache/kylin/cube/model/CubeJoinedFlatTableDesc.java [hunk 1]
```diff
@@ -66,9 +65,9 @@ public CubeJoinedFlatTableDesc(CubeSegment cubeSegment) {
     public CubeJoinedFlatTableDesc(CubeSegment cubeSegment, boolean includingDerived) {
         this(cubeSegment.getCubeDesc(), cubeSegment, includingDerived);
     }
-    
+
     private CubeJoinedFlatTableDesc(CubeDesc cubeDesc, CubeSegment cubeSegment /* can be null */,
-            boolean includingDerived) {
+                                    boolean includingDerived) {
         this.cubeDesc = cubeDesc;
```
Comment (lshmouse) [inline core-cube/src/main/java/org/apache/kylin/cube/model/CubeJoinedFlatTableDesc.java]: The format change is not expected.

File: core-metadata/src/main/java/org/apache/kylin/metadata/MetadataConstants.java [hunk 2]
```diff
@@ -52,5 +52,6 @@
     public static final String TABLE_EXD_DELIM = "delim";
     public static final String TABLE_EXD_DEFAULT_VALUE = "unknown";
 
+    @Deprecated
```
Comment (lshmouse) [inline core-metadata/src/main/java/org/apache/kylin/metadata/MetadataConstants.java]: Just remove this constant?

**Comment by lshmouse** [comment]
@bigxiaochu  Could you help to update the PR for the two small issues? Thanks~

File: core-cube/src/main/java/org/apache/kylin/cube/model/CubeJoinedFlatTableDesc.java (hunk 1 above)
Comment (bigxiaochu) [reply core-cube/src/main/java/org/apache/kylin/cube/model/CubeJoinedFlatTableDesc.java]: ok，I will fix it

File: core-metadata/src/main/java/org/apache/kylin/metadata/MetadataConstants.java (hunk 2 above)
Comment (bigxiaochu) [reply core-metadata/src/main/java/org/apache/kylin/metadata/MetadataConstants.java]: Ok. I was worried about other new code using this. I will remove this.

**Comment by bigxiaochu** [comment]
@lshmouse I hvae updated. Thranks for your review

**Comment by nichunen** [comment]
@bigxiaochu This pr is fine to me. Will you please submit another pr for the new config?

**Comment by bigxiaochu** [comment]
@nichunen pr is https://github.com/apache/kylin/pull/919

**Comment by lshmouse** [comment]
@bigxiaochu Could you help to close this PR? Thanks~

**Comment by bigxiaochu** [comment]
@lshmouse 
https://github.com/apache/kylin/pull/919 is only website document, and this pr is features code .
Because the document has been merged,  should the code  be merged  together？
Thanks

**Comment by bigxiaochu** [comment]
@nichunen  #919 is merged, could you merge this?

**Pull request merged**